activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -937,6 +937,24 @@ def build_from
       end
     end
 
+    def build_join(*args)
+      first, *rest = args
+
+      if first.is_a?(String) && rest.present?
+        sanitized = @klass.send(:sanitize_sql, args)
+
+        if sanitized == first
+          # First string doesn't contain any parameter substitution: args is a list of raw SQL
+          # strings.
+          args
```

Comment:
It's been a while since I added this. I could've sworn there was a test somewhere that passed multiple strings (which I broke), but I can't find it anymore.

Regardless, passing multiple symbols is supported. From `activerecord/lib/active_record/associations.rb`:

```
    # == Table Aliasing
    #
    # Active Record uses table aliasing in the case that a table is referenced multiple times
    # in a join. If a table is referenced only once, the standard table name is used. The
    # second time, the table is aliased as <tt>#{reflection_name}_#{parent_table_name}</tt>.
    # Indexes are appended for any more successive uses of the table name.
    #
    #   Post.joins(:comments)
    #   # => SELECT ... FROM posts INNER JOIN comments ON ...
    #   Post.joins(:special_comments) # STI
    #   # => SELECT ... FROM posts INNER JOIN comments ON ... AND comments.type = 'SpecialComment'
    #   Post.joins(:comments, :special_comments) # special_comments is the reflection name, posts is the parent table name
    #   # => SELECT ... FROM posts INNER JOIN comments ON ... INNER JOIN comments special_comments_posts
```

This seems like the cleanest solution to me. Perhaps the comment should be changed though.